Mathlib/Analysis/Equidecomposability.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2024 Chiara Cimino. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chiara Cimino, Christian Krause, Victoria Schleis
+-/
+
+import Mathlib.Analysis.NormedSpace.AffineIsometry
+import Mathlib.Order.Partition.Finpartition
+
+/-!
+Definition of Equidecomposability
+-/
+open Function
+
+variable (𝕜 : Type*) {V: Type*}(P: Type*)
+  [NormedField 𝕜]
+  [SeminormedAddCommGroup V] [NormedSpace 𝕜 V] [PseudoMetricSpace P] [NormedAddTorsor V P]
+
+/--
+Two subsets B₁ B₂ of a pseudometric space P are equidecomposable if B₁ and B₂ can be divided into
+finitely many affinely isometric parts.
+-/
+def equidecomposable (B₁ B₂ : Set P) : Prop :=
+  ∃ (P_B₁ : Finpartition B₁), ∃ (P_B₂ : Finpartition B₂), ∃ (f : P_B₁.parts → P_B₂.parts),
+    P_B₂.parts.val.sizeOf = P_B₁.parts.val.sizeOf ∧
+  Bijective f ∧ (∀ p_b₁ : P_B₁.parts, ∃ (a : AffineIsometry 𝕜 P P),
```

Comment:
The part starting with the inner `Exists` deserves to be a separate `def`.

- Do you want to require `AffineIsometry`, or just an `Isometry`?
- Do you want `a` to be bijective? If yes, then `IsometryEquiv.toRealAffineIsometryEquiv` says, that for real normed torsors the previous question is irrelevant.